src/content/get-started/data-and-json.md

```diff
@@ -0,0 +1,481 @@
+    name: _
+    publish_to: none
+
+    environment:
+      sdk: ^3.8.1 # IMPORTANT: Adjust this to match your Dart SDK version or a compatible range
+    workspace:
+      - cli
+      - command_runner
+      - wikipedia
+    ```
+
+1.  **Add workspace resolution to sub-packages.**
+
+    For each of your sub-packages (`cli`, `command_runner`, and `wikipedia`),
+    open their respective `pubspec.yaml` files and
+    add `resolution: workspace` to `pubspec.yaml`.
+    This tells Dart to resolve dependencies within the workspace.
+
+    *   For `cli/pubspec.yaml`:
+
+        ```yaml highlightLines=5
+        # ... (existing content) ...
+        name: cli
+        description: A sample command-line application.
+        version: 1.0.0
+        resolution: workspace # Add this line
+        # ... (existing content) ...
+        ```
+
+    *   For `command_runner/pubspec.yaml`:
+
+        ```yaml highlightLines=5
+        # ... (existing content) ...
+        name: command_runner
+        description: A starting point for Dart libraries or applications.
+        version: 1.0.0
+        resolution: workspace # Add this line
+        # ... (existing content) ...
+        ```
+
+    *   For `wikipedia/pubspec.yaml`:
+
+        ```yaml highlightLines=5
+        # ... (existing content) ...
+        name: wikipedia
+        description: A sample command-line application.
+        version: 1.0.0
+        resolution: workspace # Add this line
+        # ... (existing content) ...
+        ```
+
+### Task 3: Create the Summary class
+
+The Wikipedia API returns a JSON object containing a summary of an article.
+Let's create a Dart class to represent this summary.
+
+1.  Create the directory `wikipedia/lib/src/model`.
+
+    ```bash
+    mkdir -p wikipedia/lib/src/model
+    ```
+
+1.  Create the file `wikipedia/lib/src/model/summary.dart`.
+
+1.  Add the following code to `wikipedia/lib/src/model/summary.dart`:
+
+    ```dart title="wikipedia/lib/src/model/summary.dart"
+    import 'title_set.dart';
+
+    class Summary {
+      /// Returns a new [Summary] instance.
+      Summary({
+        required this.titles,
+        required this.pageid,
+        required this.extract,
+        required this.extractHtml,
+        required this.lang,
+        required this.dir,
+        this.url,
+        this.description,
+      });
+
+      ///
+      TitlesSet titles;
+
+      /// The page ID
+      int pageid;
+
+      /// First several sentences of an article in plain text
+      String extract;
+
+      /// First several sentences of an article in simple HTML format
+      String extractHtml;
+
+      /// Url to the article on Wikipedia
+      String? url;
+
+      /// The page language code
+      String lang;
+
+      /// The page language direction code
+      String dir;
+
+      /// Wikidata description for the page
+      String? description;
+
+      /// Returns a new [Summary] instance
+      static Summary fromJson(Map<String, Object?> json) {
```

Comment:
I like these `fromJson` methods are using pattern matching, but would it make sense to build them up more iteratively to explain how they are being written over a few steps? Or is it intentional that they're just given for the purposes of this article?

Definitely shouldn't block landing this though! Can be a follow up improvement.

@lamek @ericwindmill